### PR TITLE
refactor: Convert VariableAtom to Kotlin

### PR DESCRIPTION
**Pull request by david-allison**

### CODE REVIEW

**Review by lukstbit — APPROVED** [review]

File: AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt [hunk 1]
```diff
@@ -13,57 +13,35 @@
...
-    public VariableAtom(String factorString, TopLevelParser parser) {
-        this.parser = parser;
-        this.varName = factorString;
-        Pattern p = Pattern.compile("[^a-zA-Z0-9]");
-        boolean hasSpecialChar = p.matcher(varName).find();
-        if (!hasSpecialChar && varName.length() > 0) {
-            return;
+class VariableAtom(private val varName: String, private val parser: TopLevelParser) : TreeElement {
+    var atomType = AtomType.NUMBER
+    override fun getValue(): Double {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt:28]: KotlinCleanup: remove `: Double { return`.
Same below

File: AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt [hunk 2]
```diff
@@ -13,57 +13,35 @@
...
-
-        if (atomType != Atom.AtomType.INVALID) {
-
-            return parser.getVarVal(varName);
-        } else {
-            throw new ExpressionFormatException("Number is Invalid, cannot parse");
+    init {
+        val p = Pattern.compile("[^a-zA-Z0-9]")
+        val hasSpecialChar = p.matcher(varName).find()
+        if (hasSpecialChar || varName.length <= 0) {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt:43]: Clearly valid. Still, not a fan of rewritting in conversion

File: AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt:28]: It's wildplot. No need for cleanups as it's going soon

File: AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt (hunk 2 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/VariableAtom.kt:43]: Can't have a `return` in a Kotlin init block

**Pull request merged**